### ISSUE TITLE
Friendly url rewriting not working properly

### ISSUE THREAD

**Issue by johanlantz** [thirtybees/thirtybees#346]
Our current Prestashop url structure is very simple:

Basically, all urls are directly on root level after the language.
https://myshop/lang/category
https://myshop/lang/subcategory
https://myshop/lang/product

So my rewrite rules look like this:
{id}-{rewrite}

Now, since TB default URL structure is different I wanted to test this today to see that when migrating to TB my links would not break. I found some issues with this:

**Test 1: Changing Category rewrite**
*Action*: Change from an empty field to  {id}-{rewrite}
*Result*: This worked. My subcategory did now have an URL like https://myshop/lang/subcategory instead of 
https://myshop/lang/category/subcategory

**Test 2: Change back to the default TB behaviour (empty value in the field)**
*Action*: Remove my {id}-{rewrite} so it is an empty field again. Pressed Save and then Regenerate.
*Result*: Failed. My categories are still using the format https://myshop/lang/subcategory
*Expected result*: My categories links should use the default behaviour so the link is  https://myshop/lang/category/subcategory

**Test 3: Change product link rewrite to use {id}-{rewrite}**
*Action*: Change Product rewrite from empty field to {id}-{rewrite}
*Result*: Failed: All my product links stopped working. When I hoover a product image I can see that the link is the correct one so the link rewrite is ok (the link to a product does not contain the category, instead it looks like: https://myshop/lang/myProduct). However the link now takes me to a 404 page so while the link is in the format I want the actual re-direct is not ok.
*Expected result*: Clicking on a product link should take me to a Product page and not a 404.

**Test 4: Change back to the default TB behaviour (empty value in the field)**
*Action*: Remove {id}-{rewrite} so the field is again empty.
*Result*: Failed. Hoovering a product image shows the link in my custom format still (not including the category)
*Expected result*: The link format should revert to the default TB format.

The outcome is that unfortunately at the moment I can not access my products anymore even if I Save and Regenerate. So any advice on how to workaround this pending a solution for this ticket would be very welcome.

**Comment by firstred** [comment]
Thanks. We will have a look at it. It should not happen.

Do you have clean URL modules? I tested it briefly, but could not reproduce
it so far.

**Comment by johanlantz** [comment]
In the original installation I was using a SEO Pretty URL module from BVK but that one does not show up in the modules list at all after the migration.

Before creating the next clone and migration I will make sure to delete that module before starting just in case. Will let you know.

**Comment by johanlantz** [comment]
I tried launching a new TB instance on Cloudways and it behaved as expected. Something is messing with my shop after the migration. I will redo the update and migration as soon as I have some time.

One funny thing that I noticed is that on the instance I launched on Cloudways there is some info prefilled for the category and product urls while in my installation there was not.

The only difference I could think of was that my site is multilanguage so I had the option of using different rewriting rules for different languanges. I will try to test that as well using the Cloudways.

**Comment by johanlantz** [comment]
Seems to work fine with 2 languages as well. Then I guess I can only blame it on the migration. I will test from scratch when I have some time.

**Comment by firstred** [comment]
Strange. We're about to release an override checker as we've noticed that overrides are often causing strange behavior with migrations. It might help you figure things out as well.
You'll be surprised by how many are still in your store, even when uninstalling modules.

**Comment by johanlantz** [comment]
That is a good idea. Indeed some overrides from the Pretty URL module were still around even though the module was not active. The problem is that even after forcefully removing them I still could not recover working links on my page. It seemed the damage was somehow already done.

I think an override checker is an excellent idea. AFAIK the Pretty Url module is the only override I have so I will make a new clone and update Prestashop only after uninstalling the url module to see if I have more luck.

**Comment by firstred** [comment]
Hi @johanlantz,

I've gone through all the steps again, but couldn't reproduce any of the issues (did have to clear the cache a few times, though). Can you confirm it is still working for you as well, after completely removing the module's overrides and other files?

**Comment by johanlantz** [comment]
Thanks for the effort. I have not yet had time to do a complete new migration to TB.

I will get back to you once I have done that. My suspicion is that it is the Pretty URL module saving something somewhere but I really do not know. Before migrating next time, I will uninstall and delete that module and let you know what happens.

**Comment by Traumflug** [comment]
Apparently not reproducible. Please don't hesitate to re-open this issue if a reproducible case has been found.
